richard juckes:
patch by Alan Modra to allow successful build of binutils with gcc-4.x

http://bugs.gentoo.org/show_bug.cgi?id=82907
http://sources.redhat.com/ml/binutils/2005-02/msg00393.html

diff -ur binutils-2.15.92.0.2-orig/gas/as.h binutils-2.15.92.0.2/gas/as.h
--- binutils-2.15.92.0.2-orig/gas/as.h	2005-03-04 02:47:28.000000000 -0500
+++ binutils-2.15.92.0.2/gas/as.h	2005-03-04 02:49:24.000000000 -0500
@@ -397,7 +397,23 @@
 /* Enough bits for address, but still an integer type.
    Could be a problem, cross-assembling for 64-bit machines.  */
 typedef addressT relax_addressT;
-
+
+struct relax_type
+{
+    /* Forward reach. Signed number. > 0.  */
+      offsetT rlx_forward;
+    /* Backward reach. Signed number. < 0.  */
+      offsetT rlx_backward;
+  
+      /* Bytes length of this address.  */
+      unsigned char rlx_length;
+  
+      /* Next longer relax-state.  0 means there is no 'next' relax-state.  */
+      relax_substateT rlx_more;
+  };
+
+typedef struct relax_type relax_typeS;
+
 /* main program "as.c" (command arguments etc).  */
 
 COMMON unsigned char flag_no_comments; /* -f */
diff -ur binutils-2.15.92.0.2-orig/gas/tc.h binutils-2.15.92.0.2/gas/tc.h
--- binutils-2.15.92.0.2-orig/gas/tc.h	2005-03-04 02:47:28.000000000 -0500
+++ binutils-2.15.92.0.2/gas/tc.h	2005-03-04 02:48:26.000000000 -0500
@@ -24,25 +24,6 @@
 
 extern const pseudo_typeS md_pseudo_table[];
 
-/* JF moved this here from as.h under the theory that nobody except MACHINE.c
-   and write.c care about it anyway.  */
-
-struct relax_type
-{
-  /* Forward reach. Signed number. > 0.  */
-  long rlx_forward;
-  /* Backward reach. Signed number. < 0.  */
-  long rlx_backward;
-
-  /* Bytes length of this address.  */
-  unsigned char rlx_length;
-
-  /* Next longer relax-state.  0 means there is no 'next' relax-state.  */
-  relax_substateT rlx_more;
-};
-
-typedef struct relax_type relax_typeS;
-
 extern const int md_reloc_size;	/* Size of a relocation record.  */
 
 char * md_atof (int, char *, int *);
